Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -456,6 +484,109 @@ public int compare(String text1, String text2) {
+                                break;
+                            case "lastModified":
+                                note.setModificationDate(DateTimeUtils.getDateCalendar(noteJson.getString(key)));
+                                break;
+                            case "tags":
+                                note.setTags(jsonToList(noteJson.getJSONArray(key)));
+                                break;
+                            case "pinned":
+                                note.setPinned(noteJson.getBoolean(key));
+                                break;
+                            case "markdown":
+                                note.setMarkdownEnabled(noteJson.getBoolean(key));
+                                break;
+                            case "publicURL":
+                                // note.setPublished(?);
+                                break;
+                        }
+                    }
+                    note.save();
+                }
+            } else {
+                Note note = noteBucket.newObject();
+                note.setContent(noteContent);
+                note.setCreationDate(Calendar.getInstance());
+                note.setModificationDate(note.getCreationDate());
+                note.save();
+            }
+            Toast.makeText(requireContext(), getString(R.string.import_message_success), Toast.LENGTH_SHORT).show();
+        } catch (Exception e) {
+            Toast.makeText(requireContext(), getString(R.string.import_message_failure), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private List<String> jsonToList(JSONArray json) {
+        List<String> list = new ArrayList<>();
+        for (int i = 0; i < json.length(); i++) {
+            try {
+                list.add(json.getString(i));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+        return list;
+    }
+
+    private String readFile(Uri uri) throws IOException {
+        StringBuilder stringBuilder = new StringBuilder();
+        InputStream inputStream = requireContext().getContentResolver().openInputStream(uri);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        String line;
+        while ((line = reader.readLine()) != null) {
+            stringBuilder.append(line);
+            stringBuilder.append("\n");
+        }
+        inputStream.close();
+        return stringBuilder.toString();
+    }
+
+    public String getFileExtension(Uri uri) {
+        String extension;
+        if (uri.getScheme().equals(ContentResolver.SCHEME_CONTENT)) {
+            final MimeTypeMap mime = MimeTypeMap.getSingleton();
+            extension = mime.getExtensionFromMimeType(requireContext().getContentResolver().getType(uri));
+        } else {
+            extension = MimeTypeMap.getFileExtensionFromUrl(Uri.fromFile(new File(uri.getPath())).toString());
```

Comment:
does Android do anything to more accurately compute the MIME type from a document? if that's the case, if it does more than look at the extension, would it be more helpful if we used that vs. relying on the extension? or a combination of both?

Reply:
Yes, I think I found a better solution, I'll fix it!